webrender/src/renderer.rs

```diff
@@ -5043,6 +5059,47 @@ impl Renderer {
         self.external_image_handler = Some(Box::new(image_handler) as Box<_>);
         info!("done.");
     }
+
+    fn create_texture(
+        &mut self,
+        target: TextureTarget,
+        format: ImageFormat,
+        width: u32,
+        height: u32,
+        filter: TextureFilter,
+        render_target: Option<RenderTargetInfo>,
+        layer_count: i32,
+    ) -> Texture {
+        let alloc = self.device.create_texture(
+            target,
+            format,
+            width,
+            height,
+            filter,
+            render_target,
+            layer_count,
+        );
+
+        match alloc {
+            Ok(texture) => texture,
+            Err(err) => {
+                self.log_allocation_failure(err);
+                panic!();
+            }
+        }
+    }
+
+    fn log_allocation_failure(&self, error: AllocError) {
```

Comment:
Since the budget is there to force a crash when things are so wrong that it's best not to keep the browser alive, I don't think it is very interesting to display.
That said I think that displaying some number that we think is reasonable for an average page, or some number beyond which we believe something worth investigating (but not to the point that we'd crash the browser) would be useful, as well as displaying the current allocated texture memory.